### DISCUSSION TITLE
batch_run troubles

### DISCUSSION THREAD

**Discussion by AnnedeKoeijer** [mesa/mesa#1802] [Q&A]
I am having some troubles collecting my data through the batch_run function. It seems as if this function is not returning the data of the last timestep but the second to last timestep. I do not think there is a data collection coding issue in my model because when I run the model in a normal way once and collect its data (thorugh model.datacollector.get_model_vars_dataframe()), it does provide me with a.o. the results of the last time step.
I have been racking my brain over a few hours now and I still am not entirely certain if it has something to do with my code or just the batch_run function. Are other people experiencing this issue aswell?

**Comment by rht** [comment]
Do you have a minimal reproducer? Any of the models in https://github.com/projectmesa/mesa-examples/tree/main/examples can be a starting point.
